packages/location-field/src/index.tsx

```diff
@@ -0,0 +1,758 @@
+
+// FIXME have a better key generator for options
+let optionKey = 0;
+
+function DefaultLocationIcon({
+  locationType
+}: {
+  locationType: string;
+}): React.ReactElement {
+  return <LocationIcon size={13} type={locationType} />;
+}
+
+const LocationField = ({
+  autoFocus = false,
+  addLocationSearch = () => {},
+  className = null,
+  clearLocation = () => {},
+  clearButtonIcon = <Times size={13} />,
+  currentPosition = null,
+  currentPositionIcon = <LocationArrow size={13} />,
+  currentPositionUnavailableIcon = <Ban size={13} />,
+  initialSearchResults = null,
+  preferredLayers = [],
+  findNearbyStops = () => {},
+  getCurrentPosition,
+  GeocodedOptionIconComponent = GeocodedOptionIcon,
+  geocoderConfig,
+  hideExistingValue = false,
+  inputPlaceholder = null,
+  layerColorMap = {},
+  location = null,
+  locationType,
+  LocationIconComponent = DefaultLocationIcon,
+  nearbyStops = [],
+  onTextInputClick = null,
+  operatorIconMap = {},
+  sessionOptionIcon = <Search size={13} />,
+  sessionSearches = [],
+  sortByDistance = false,
+  showClearButton = true,
+  onLocationSelected,
+  showUserSettings = false,
+  isStatic = false,
+  stopOptionIcon = <Bus size={13} />,
+  stopsIndex = null,
+  suppressNearby = false,
+  suggestionCount = 3,
+  userLocationsAndRecentPlaces = [],
+  UserLocationIconComponent = UserLocationIcon
+}: LocationFieldProps): React.ReactElement => {
+  /**
+   * Gets the initial value to place in the input field.
+   */
+  const getValueFromLocation = () => {
+    const label = location?.name || "";
+    return location && !hideExistingValue ? label : "";
+  };
+
+  const listBoxId = `listbox-${optionKey}`;
+
+  const intl = useIntl();
+
+  const [activeIndex, setActiveIndex] = useState(null);
+  const [stateGeocodedFeatures, setGeocodedFeatures] = useState([]);
+  const [menuVisible, setMenuVisible] = useState(false);
+  const [stateMessage, setMessage] = useState(null);
+  const [stateValue, setValue] = useState(getValueFromLocation());
+
+  const inputRef = useRef(null);
+
+  useEffect(() => {
+    // location could be null if none is set
+    setValue(location?.name || "");
+    setGeocodedFeatures([]);
+  }, [location]);
+
+  useEffect(() => {
+    if (initialSearchResults) {
+      setGeocodedFeatures(initialSearchResults);
+      setMenuVisible(true);
+    }
+  }, [initialSearchResults]);
+
+  const geocodeAutocomplete = throttle(1000, text => {
+    if (!text) {
+      console.warn("No text entry provided for geocode autocomplete search.");
+      return;
+    }
+    getGeocoder(geocoderConfig)
+      .autocomplete({ text })
+      // TODO: Better type?
+      .then(
+        (result: {
+          features: any;
+          results: { error: { message: string } };
+        }) => {
+          let message: string;
+          // If no features found in response, default to empty array.
+          let geocodedFeatures = result && result.features;
```

Comment:
Use `?.` syntax.

Reply:
Addressed in 18d5606109e7dfce7d00d42b42c8b1b6f075a0bc